addon/components/qr-scanner.js

```diff
@@ -85,6 +90,26 @@ export default Ember.Component.extend({
     if (timerId) {
       Ember.run.cancel(timerId);
     }
+
+    // Cross browser stream.stop hack from:https://stackoverflow.com/questions/11642926/stop-close-webcam-which-is-opened-by-navigator-getusermedia
+    var MediaStream = window.MediaStream;
+    if (typeof MediaStream === 'undefined' && typeof webkitMediaStream !== 'undefined') {
+        MediaStream = webkitMediaStream;
+    }
+    if (typeof MediaStream !== 'undefined' && !('stop' in MediaStream.prototype)) {
+        MediaStream.prototype.stop = function() {
+            this.getAudioTracks().forEach(function(track) {
+                track.stop();
+            });
+
+            this.getVideoTracks().forEach(function(track) {
+                track.stop();
+            });
+        };
+    }
```

Comment:
there is no need to apply this hack every time the component is destroyed. if we really need this we should add it to the vendor folder and only run it once on app boot.